design/ceph/object/ceph-bucket-notification-crd.md

```diff
@@ -0,0 +1,82 @@
+# Ceph bucket notifications CRD
+
+## Overview
+
+Ceph added support for the bucket notifications feature from Nautilus onwards. It allows to send messages to various endpoints when a new event occurs on a bucket (ref: <https://docs.ceph.com/docs/master/radosgw/notifications/)>
+
+Setup of those notifications are normally done by sending HTTP requests to the RGW, either to create/delete topics pointing to specific endpoints, or create/delete bucket notifications based on those topics.
+
+This functionality eases this process by avoiding to use external tools or scripts. It is replaced by creation of CR definitions that contain all the information necessary to create topics and/or notifications, which the rook operator processes.
+
+## Goals
+
+Creates a CRD for topics and a CRD for notifications, defining all the necessary and optional information for the various endpoints.
+
+Extends the rook operator to handle the CRs that would be submitted by users.
+
+## Implementation
+
+The CR for a topic configuration takes this form:
+
+```yaml
+apiVersion: ceph.rook.io/v1
+kind: CephBucketTopic
```

Comment:
How about adding sample values to help understand the settings? It's hard to tell even if they are strings, bools, etc.